climada_petals/entity/impact_funcs/river_flood.py

```diff
@@ -55,154 +53,452 @@ def __init__(self):
-        elif region.lower() == "europe":
-            impf.id = 3
-            impf.name = "Flood Europe JRC Residential noPAA"
-            impf.continent = 'Europe'
-            impf.intensity = np.array([0., 0.5, 1., 1.5, 2., 3., 4., 5., 6., 12.])
-            impf.mdd = np.array([0.00, 0.25, 0.40, 0.50, 0.60, 0.75, 0.85, 0.95,
-                                 1.00, 1.00])
-            impf.mdr = np.array([0.000, 0.250, 0.400, 0.500, 0.600, 0.750, 0.850,
-                                 0.950, 1.000, 1.000])
-        elif region.lower().replace(" ", "") == "northamerica":
-            impf.id = 4
-            impf.name = "Flood North America JRC Residential noPAA"
-            impf.continent = 'NorthAmerica'
-            impf.intensity = np.array([0., 0.1, 0.5, 1., 1.5, 2., 3., 4., 5.,
-                                      6., 12.])
-            impf.mdd = np.array([0.0000, 0.2018, 0.4433, 0.5828, 0.6825, 0.7840,
-                                 0.8543, 0.9237, 0.9585, 1.0000, 1.0000])
-            impf.mdr = np.array([0.0000, 0.2018, 0.4433, 0.5828, 0.6825, 0.7840,
-                                 0.8543, 0.9237, 0.9585, 1.0000, 1.0000])
-        elif region.lower() == "oceania":
-            impf.id = 5
-            impf.name = "Flood Oceania JRC Residential noPAA"
-            impf.continent = 'Oceania'
-            impf.intensity =  np.array([0., 0.5, 1., 1.5, 2., 3., 4., 5., 6., 12.])
-            impf.mdd = np.array([0.00, 0.48, 0.64, 0.71, 0.79, 0.93, 0.97, 0.98,
-                                 1.00, 1.00])
-            impf.mdr = np.array([0.000, 0.480, 0.640, 0.710, 0.790, 0.930, 0.970,
-                                 0.980, 1.000, 1.000])
-        elif region.lower().replace(" ", "") == "southamerica":
-            impf.id = 6
-            impf.name = "Flood South America JRC Residential noPAA"
-            impf.continent = 'SouthAmerica'
-            impf.intensity =  np.array([0., 0.5, 1., 1.5, 2., 3., 4., 5., 6., 12.])
-            impf.mdd = np.array([0.0000, 0.4908, 0.7112, 0.8420, 0.9494,
-                                 0.9836, 1.0000, 1.0000, 1.0000, 1.0000])
-            impf.mdr = np.array([0.0000, 0.4908, 0.7112, 0.8420, 0.9494, 0.9836,
-                                 1.0000, 1.0000, 1.0000, 1.0000])
+
+        if sector == 'residential':
+            impf_values, impf_id = from_jrc_impf_residential(region)
+
+        elif sector == 'industrial':
+            impf_values, impf_id = from_jrc_impf_industrial(region)
+
+        elif sector == 'commercial':
+            impf_values, impf_id = from_jrc_impf_commercial(region)
+
+        elif sector == 'transport':
+            impf_values, impf_id = from_jrc_impf_transport(region)
+
+        elif sector == 'infrastructure':
+            impf_values, impf_id = from_jrc_impf_infrastructure(region)
+
+        elif sector == 'agriculture':
+            impf_values, impf_id = from_jrc_impf_agriculture(region)
+
```

Comment:
Long chains of `if`  and `else` statements are to be avoided. For future work please use a dataclass or similar.